src/SignalR.Orleans/Extensions.cs

```diff
@@ -1,54 +1,30 @@
 using Microsoft.AspNetCore.SignalR;
 using Orleans;
 using Orleans.Hosting;
-using Orleans.Runtime.Configuration;
 using SignalR.Orleans;
-using SignalR.Orleans.Clients;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
     public static class OrleansServerExtensions
     {
-        public static ClusterConfiguration AddSignalR(this ClusterConfiguration config)
-        {
-            config.AddSimpleMessageStreamProvider(Constants.STREAM_PROVIDER);
-            try
-            {
-                config.AddMemoryStorageProvider("PubSubStore");
-            }
-            catch
-            {
-                // PubSubStore was already added. Do nothing.
-            }
-            config.AddMemoryStorageProvider(Constants.STORAGE_PROVIDER);
-            return config;
-        }
-
         public static ISiloHostBuilder UseSignalR(this ISiloHostBuilder builder)
         {
-            return builder.ConfigureApplicationParts(mgr =>
-                mgr.AddApplicationPart(typeof(ClientGrain).Assembly));
+            try { builder = builder.AddMemoryGrainStorage("PubSubStore"); }
+            catch { /** PubSubStore was already added. Do nothing. **/ }
+
+            return builder.AddMemoryGrainStorage(Constants.STORAGE_PROVIDER)
+                .AddSimpleMessageStreamProvider(Constants.STREAM_PROVIDER)
+                .ConfigureApplicationParts(parts => parts.AddFromAppDomain().AddFromApplicationBaseDirectory());
```

Comment:
Please revert this line to the previous one. Load from appdomain/base directory impose a restriction. 

Load the current assembly is the correct way if doing it.

Reply:
is it ok if i will do it like this:
.ConfigureApplicationParts(parts => parts.AddApplicationPart(Assembly.GetExecutingAssembly()).WithReferences())

instead of specifing an actual class:
 .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(IClientGrain).Assembly).WithReferences())

Reply:
Executing assembly is where the Main() entrypoint is. The way it is, we ensure the assembly is loaded properly to avoid problems of assembly loading/binding at runtime in Orleans.